TKGame/Animations/ScreenTransistion.cs

```diff
@@ -0,0 +1,108 @@
+            sourceRectangles[6] = new Rectangle(buffer, (spriteY *2) + (buffer), spriteX, spriteY);
+            sourceRectangles[7] = new Rectangle((spriteX + buffer) + 2, (spriteY + (buffer * 2)) * 2, spriteX, spriteY);
+            sourceRectangles[8] = new Rectangle((spriteX * 2) + (buffer * 2), (spriteY + (buffer * 2)) * 2, spriteX, spriteY);
+            sourceRectangles[9] = new Rectangle(buffer, (spriteY * 3) + (2 * buffer) , spriteX, spriteY);
+            sourceRectangles[10] = new Rectangle((spriteX + buffer) + 2, (spriteY * 3) + (2 * buffer), spriteX, spriteY);
+            sourceRectangles[11] = new Rectangle((spriteX * 2) + (buffer * 2), (spriteY * 3) + (6 * buffer), spriteX, spriteY);
+            sourceRectangles[12] = new Rectangle(buffer, (spriteY + buffer) * 4, spriteX, spriteY);
+            sourceRectangles[13] = new Rectangle((spriteX + buffer) + 2, (spriteY + buffer) * 4, spriteX, spriteY);
+            sourceRectangles[14] = new Rectangle((spriteX * 2) + (buffer * 2), (spriteY + buffer) * 4, spriteX, spriteY);
+            sourceRectangles[15] = new Rectangle(buffer, (spriteY + buffer) * 5, spriteX, spriteY);
+            sourceRectangles[0] = new Rectangle((spriteX + buffer) + 2, (spriteY +buffer) * 5 + 16, spriteX, spriteY);
+
+        }
+        /// <summary>
+        /// Draws Screen Transition Rectangle the size of game window
+        /// </summary>
+        /// <param name="spriteBatch"></param>
+        public void Draw(SpriteBatch spriteBatch)
+        {
+            //Draw a rectangle the size of background
+            spriteBatch.Draw(Art.LoadTexture, new Rectangle(0, 0, screenWidth, screenHeight), sourceRectangles[currentIndex], Color.White);
+        }
+
+        /// <summary>
+        /// Updates transition spritesheet through different images 
+        /// </summary>
+        /// <param name="gameTime"></param>
+        public void Update(GameTime gameTime)
+        {
+            if (timer > threshold + 500) //waits for about half a second and then resets the loop 
+                loop = true;
+
+                if (!loop)
+                {
+                    timer += (float) gameTime.ElapsedGameTime.TotalMilliseconds; //just increments timer and leaves
+                    return;
+                }
+                //logic to loop through images
+                if (timer > threshold) //checks if enough time has passed
+                {
+                    if (currentIndex < 15)
+                        currentIndex++; //increments through sprite sheet
+                    else
+                    {
+                        currentIndex = 0; // sets image back to full transparent window
+                        loop = false; //stops incrementing at end of sprite sheet
+                    }
+                    timer = 0; // reset timer
+                }
+                else
+                    timer += (float)gameTime.ElapsedGameTime.TotalMilliseconds; //increases timer
```

Comment:
This does what the if(!loop) condition does but actually gets run

Reply:
the if(!loop) is just there to stop it from running the rest of the code to increment the sprite sheet, this just has the timer increment if it is not greater than the threshold, which is 50ms